apps/crashing.go

```diff
@@ -45,6 +47,69 @@ var _ = AppsDescribe("Crashing", func() {
 		})
 	})
 
+	Describe("an app with three instances, two running and one crashing", func() {
+		It("keeps two instances running while another crashes", func() {
+			By("Pushing the app with three instances")
+			Expect(cf.Cf(
+				"push", appName,
+				"-b", "python_buildpack",
+				"-m", DEFAULT_MEMORY_LIMIT,
+				"-p", assets.NewAssets().PythonCrashApp,
+				"-i", "3", // Setting three instances
+			).Wait(Config.CfPushTimeoutDuration())).To(Exit(0))
+
+			By("Checking that the app is up and running")
+			Eventually(cf.Cf("app", appName)).Should(Say("running"))
+
+			By("Waiting until one instance crashes")
+			appGuid := app_helpers.GetAppGuid(appName)
+			processStatsPath := fmt.Sprintf("/v3/apps/%s/processes/web/stats", appGuid)
+
+			// Poll until at least one instance has crashed
+			Eventually(func() bool {
+				session := cf.Cf("curl", processStatsPath).Wait()
```

Comment:
Could this block be extracted into a function? Then you could call
```
hasOneInstanceState("CRASHED")
...
hasOneInstanceState("RUNNING")
```

Reply:
Good catch